backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/BitcodeEmbedding.kt

```diff
@@ -17,31 +16,16 @@ object BitcodeEmbedding {
     }
 
     internal fun getClangOptions(config: KonanConfig): List<String> = when (config.bitcodeEmbeddingMode) {
-        BitcodeEmbedding.Mode.NONE -> listOf("-fembed-bitcode=off")
-        BitcodeEmbedding.Mode.MARKER -> listOf("-fembed-bitcode=marker")
-        BitcodeEmbedding.Mode.FULL -> listOf("-fembed-bitcode=all")
-    }
-
-    private val KonanConfig.bitcodeEmbeddingMode get() = configuration.get(KonanConfigKeys.BITCODE_EMBEDDING_MODE)!!.also {
-        require(it == Mode.NONE || this.produce == CompilerOutputKind.FRAMEWORK) {
-            "${it.name.toLowerCase()} bitcode embedding mode is not supported when producing ${this.produce.name.toLowerCase()}"
-        }
+        Mode.NONE -> listOf("-fembed-bitcode=off")
+        Mode.MARKER -> listOf("-fembed-bitcode=marker")
+        Mode.FULL -> listOf("-fembed-bitcode=all")
     }
 
-    internal fun processModule(llvm: Llvm) = when (llvm.context.config.bitcodeEmbeddingMode) {
-        Mode.NONE -> {}
-        Mode.MARKER -> {
-            addEmptyMarker(llvm, "konan_llvm_bitcode", "__LLVM,__bitcode")
-            addEmptyMarker(llvm, "konan_llvm_cmdline", "__LLVM,__cmdline")
+    private val KonanConfig.bitcodeEmbeddingMode: Mode
+        get() = configuration.get(KonanConfigKeys.BITCODE_EMBEDDING_MODE)!!.also {
+            // TODO: We cannot produce programs due to `-alias` linked flag.
```

Comment:
Hmm, so why not fix alias with LLVM API call as suggested in https://github.com/JetBrains/kotlin-native/pull/3288?